pkg/audit/verifier.go

```diff
@@ -251,12 +298,44 @@ func (verifier *Verifier) Reverify(ctx context.Context, stripe *Stripe) (report
 
 			share, err := verifier.GetShare(ctx, limit, pending.StripeIndex, pending.ShareSize, int(piece.PieceNum))
 			if err != nil {
-				// TODO(kaloyan): we need to check the logic here if we correctly identify offline nodes from those that didn't respond.
-				if err == context.DeadlineExceeded || !transport.Error.Has(err) {
+				if transport.Error.Has(share.Error) {
+					if errs.IsFunc(share.Error, func(err error) bool {
+						return err == context.DeadlineExceeded
+					}) {
+						// dial timeout
+						ch <- result{nodeID: piece.NodeId, status: offline}
+						return
+					}
+					if errs.IsFunc(share.Error, func(err error) bool {
+						return status.Code(err) == codes.Unknown
+					}) {
+						// dial failed -- offline node
+						ch <- result{nodeID: piece.NodeId, status: offline}
+						return
+					}
+					// unknown transport error
 					ch <- result{nodeID: piece.NodeId, status: contained}
-				} else {
-					ch <- result{nodeID: piece.NodeId, status: offline}
+					return
+				}
+
+				if errs.IsFunc(share.Error, func(err error) bool {
+					return status.Code(err) == codes.NotFound
+				}) {
+					// missing share
+					ch <- result{nodeID: piece.NodeId, status: failed}
+					return
+				}
+
+				if errs.IsFunc(share.Error, func(err error) bool {
+					return status.Code(err) == codes.DeadlineExceeded
+				}) {
+					// dial successful, but download timed out
+					ch <- result{nodeID: piece.NodeId, status: contained}
```

Comment:
found through testing that for all cases where status is contained, the additional field `pendingAudit` needs to be populated with `pending`.

Reply:
actually I'll just add the fix

Reply:
You actually found the reason for the panic Jens sees! Thank you!